### PR TITLE
docs/CONTRIBUTING: Update Polkadot pull request process documentation

### PR DESCRIPTION
**Pull request by mxinden**
As far as I can tell one does not need to wait for both the Substrate and the Polkadot pull request to go green, but only for the former to go green. Once that happens the bot takes care of updating the Substrate reference on the Polkadot pull request. This pull request updates the docs accordingly.

This is **my understanding** of how this process is supposed to work. Can someone with a deeper understanding of the process review the documentation update?

### CODE REVIEW

File: docs/CONTRIBUTING.adoc [hunk 1]
```diff
@@ -71,8 +71,15 @@ To create a Polkadot companion PR:
 . Make the changes required and build polkadot locally.
 . Submit all this as a PR against the Polkadot Repo. Link to your Polkadot PR in the _description_ of your Substrate PR as "polkadot companion: [URL]"
 . Now you should see that the `check_polkadot` CI job will build your Substrate PR agains the mentioned Polkadot branch in your PR description.
-. Wait for reviews on both
-. Once both PRs have been green lit, they can both be merged 🍻.
+. Wait for reviews on both the Substrate and the Polkadot pull request.
+. Once the Substrate pull request has been green lit, one can comment on the Substrate pull request with `bot merge` which will:
```
Comment (bkchr) [inline docs/CONTRIBUTING.adoc:75]: "bot merge" is only doable by members of the parity group in github.

Comment (dvdplm) [inline docs/CONTRIBUTING.adoc:75]:
```suggestion
. Once the Substrate pull request runs green, a member of the `parity` github group can comment on the Substrate pull request with `bot merge` which will:
```

**Review by dvdplm — APPROVED** [review]

**Comment by bkchr** [comment]
@mxinden please merge master.

**Comment by mxinden** [comment]
bot merge

**Comment by mxinden** [comment]
> The PR has approval from a member of substrateteamleads.

@bkchr as a Substrate team lead, would you mind merging this pull request?

**Review by bkchr — APPROVED** [review]

**Pull request merged**